packages/tangle/approvalweightmanager.go

```diff
@@ -86,6 +87,70 @@ func (a *ApprovalWeightManager) weightsPerEpoch(branchID ledgerstate.BranchID) *
 	return weightsPerEpoch
 }
 
+func (a *ApprovalWeightManager) onMarkerConfirmed(marker *markers.Marker, messageID MessageID) {
+	// walk through the marker's pastcone to update FinalizedApprovalWeight
+	finalizedWalker := walker.New()
+
+	// push marker itself
+	finalizedWalker.Push(messageID)
+	a.tangle.Storage.MessageMetadata(messageID).Consume(func(metadata *MessageMetadata) {
+		metadata.SetFinalizedApprovalWeight(true)
+	})
+	// push parents, keep walking
+	a.tangle.Storage.Message(messageID).Consume(func(message *Message) {
+		message.ForEachParent(func(parent Parent) {
```

Comment:
I'm not sure if we can propagate this over all parents. Strong parents for sure, but weak parents might need a special treatment?

Reply:
yes, we need to discuss this. I'll replace it with strong parents for now.

Reply:
for weak parent: mark finalized, but don't propagate.